e2e/preferences.e2e.spec.ts

```diff
@@ -0,0 +1,142 @@
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: reportAsset(__filename) });
+    await packageLogs(__filename);
+    await electronApp.close();
+  });
+
+  test('should open preferences modal', () => {
+    expect(preferencesWindow).toBeDefined();
+  });
+
+  test('should show application page', async() => {
+    const { application } = new PreferencesPage(preferencesWindow);
+
+    await expect(application.nav).toHaveClass('preferences-nav-item active');
+
+    if (!os.platform().startsWith('win')) {
+      await expect(application.tabBehavior).toHaveText('Behavior');
+      await expect(application.administrativeAccess).toBeVisible();
+    }
+
+    await expect(application.automaticUpdates).toBeVisible();
+    await expect(application.statistics).toBeVisible();
+    await expect(application.pathManagement).not.toBeVisible();
+  });
+
+  test('should render environment tab', async() => {
+    test.skip(os.platform() === 'win32', 'Environment tab not available on Windows');
+    const { application } = new PreferencesPage(preferencesWindow);
+
+    application.tabEnvironment.click();
+
+    await expect(application.administrativeAccess).not.toBeVisible();
+    await expect(application.automaticUpdates).not.toBeVisible();
+    await expect(application.statistics).not.toBeVisible();
+    await expect(application.pathManagement).toBeVisible();
+  });
+
+  test('should not render tabs in windows', async() => {
+    test.skip(os.platform() !== 'win32', 'Environment and behavior tabs exist on macOS & Linux');
+
+    const { application } = new PreferencesPage(preferencesWindow);
+
+    await expect(application.tabBehavior).not.toBeVisible();
+    await expect(application.tabEnvironment).not.toBeVisible();
+  });
+
+  test('should navigate to virtual machine', async() => {
+    test.skip(os.platform() === 'win32', 'Virtual Machine not available on Windows');
+    const { virtualMachine, application } = new PreferencesPage(preferencesWindow);
+
+    virtualMachine.nav.click();
+
+    await expect(application.nav).toHaveClass('preferences-nav-item');
+    await expect(virtualMachine.nav).toHaveClass('preferences-nav-item active');
+    await expect(virtualMachine.memory).toBeVisible();
+    await expect(virtualMachine.cpus).toBeVisible();
+  });
+
+  test('should navigate to container runtime', async() => {
+    const { containerRuntime } = new PreferencesPage(preferencesWindow);
+
+    containerRuntime.nav.click();
+
+    await expect(containerRuntime.nav).toHaveClass('preferences-nav-item active');
+    await expect(containerRuntime.containerRuntime).toBeVisible();
+  });
+
+  test('should navigate to kubernetes', async() => {
+    const { kubernetes, containerRuntime } = new PreferencesPage(preferencesWindow);
+
+    kubernetes.nav.click();
+
+    await expect(containerRuntime.nav).toHaveClass('preferences-nav-item');
+    await expect(kubernetes.nav).toHaveClass('preferences-nav-item active');
```

Comment:
One of these consistently fail for me (but it's not always the same one), even after `await`ing for the click result.  I'm not sure what's going on — the click is supposed to wait for any resulting navigation.
I'm also not sure why these, and not any of the other similar tests.

Reply:
I'm not experiencing this in any of my environments, but I'm seeing test timeouts in CI so something might be amiss.. I'll take a closer look and see if there's anything I can't uncover.

Reply:
It seems to work for me now, so 🤷